### ISSUE TITLE
Support Oracle AQ as queue provider

### ISSUE THREAD

**Issue by fbascheper** [jmstoolbox/jmstoolbox#81]
## Specifications
- JMSToolBox version:
- OS (Name, version, 32/64bits):
- Q provider (Oracle AQ 19.3.0.0) :

## Expected Behavior
It should be possible to select Oracle AQ In the Q Managers tab.

## Actual Behavior
Only Oracle WebLogic provider is available, but it's not possible to use Oracle AQ directly.

## Steps to Reproduce the Problem
N/A

## Logs
N/A

---

Now that the relevant Maven artifacts have been published on Maven Central, it should be possible to add Oracle AQ as a supported Queue Provider. In my current project, we are using the following dependencies:

```xml
<dependency>
   <groupId>com.oracle.database.jdbc</groupId>
   <artifactId>ojdbc8</artifactId>
   <version>19.3.0.0</version>
   <exclusions>
       <exclusion>
           <groupId>com.oracle.database.ha</groupId>
           <artifactId>*</artifactId>
       </exclusion>
       <exclusion>
           <groupId>com.oracle.database.security</groupId>
           <artifactId>osdt_cert</artifactId>
       </exclusion>
       <exclusion>
           <groupId>com.oracle.database.security</groupId>
           <artifactId>osdt_core</artifactId>
       </exclusion>
   </exclusions>
</dependency>
<dependency>
   <groupId>com.oracle.database.messaging</groupId>
   <artifactId>aqapi</artifactId>
   <version>19.3.0.0</version>
</dependency>
```

**Comment by fbascheper** [comment]
BTW I can probably help to fix this issue, if you give me a few pointers where to start and how to integrate and test changes.

**Comment by titou10titou10** [comment]
Cool. PR are welcome!
You can find basic instructions to setup a dev environment [here](https://github.com/jmstoolbox/jmstoolbox/blob/dev/org.titou10.jtb.core/doc/setup_eclipse.txt)
Basically, each QM is managerd by an eclipse plugin that must extend [this class](https://github.com/jmstoolbox/jmstoolbox/blob/dev/org.titou10.jtb.core/src/org/titou10/jtb/jms/qm/QManager.java)
The JMS standard does not define a way to retrieve the list of JMS objects from a provider, (as opposed to JDBC for example) so the main difficulty is usually to find a way for JTB to retrieve the list of Queues and Topics. Some QM have specific interfaces for that, some usie JNDI, other JMX etc.
Also it hs to be 100% JMS compliant
After those subjects are sorted out, I can create a plugin for it in a branch in JTB

Q:
- do you have previous experience in eclipse plugin dev?
- What is the licensing of  Oracle AQ? is it a FOSS? Is there a way for me to install a test environment somewhere?
- do you know how to retrieve the list of Q and topics from a Java client?

**Comment by titou10titou10** [comment]
https://stackoverflow.com/questions/62102276/oracle-aq-is-there-a-way-to-retrieve-the-list-of-queues-and-topics-from-a-java

**Comment by titou10titou10** [comment]
Implemented in v5.16.0
